src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/Services/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@ContextConfiguration(classes = {AuthenticationService.class})
+@ExtendWith(SpringExtension.class)
+public class AuthenticationServiceTest {
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    ResponseEntity<String> response401;
+    ResponseEntity<String> response403;
+    ResponseEntity<String> lastError;
+
+    User user;
+
+    @BeforeEach
+    void setUp() {
+        response401 = ResponseEntity.status(401).body("Unauthorized, missing authentication.");
+        response403 = ResponseEntity.status(403).body("Forbidden.");
+        lastError = null;
+
+        user = new User();
+    }
+
+    @Test
+    void testAuthentication() {
+        assertFalse(authenticationService.authentication("John"));
+        assertTrue(authenticationService.authentication("passing"));
+        assertTrue(authenticationService.authentication("admin"));
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualNull() {
+        assertFalse(authenticationService.authentication(null));
+        lastError = response401;
+        assertEquals(response401,lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualAdmin() {
+        assertTrue(authenticationService.authentication("admin"));
+        assertNull(lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualPassing() {
+        assertTrue(authenticationService.authentication("passing"));
+        assertNull(lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualSomethingElse() {
+        assertFalse(authenticationService.authentication("John"));
+        lastError = response403;
+        assertEquals(response403,lastError);
+    }
+
+    @Test
+    void testGetError() {
+        assertNull((new AuthenticationService()).getError());
+    }
+
+    @Test
+    void testLoginWithSuccessfulLogin() {
+        user.setEmail("test@fau.de");
+        user.setId(1);
+        user.setName("Name0");
+        user.setPassword("test");
+        user.setSurname("Surname0");
+        user.setUserRole(UserRole.ADMIN);
+
+        Optional<User> ofResult = Optional.of(user);
+        when(userRepository.findByEmail(any())).thenReturn(ofResult);
+        ResponseEntity<String> actualLoginResult = authenticationService.login("test@fau.de", "test");
+
+        assertEquals("\"Login successful.\"", actualLoginResult.getBody());
+        assertEquals(200, actualLoginResult.getStatusCodeValue());
+        assertTrue(actualLoginResult.getHeaders().isEmpty());
+        verify(userRepository).findByEmail(any());
+    }
+
+    @Test
+    void testLoginWithWrongPassword() {
+        User user = mock(User.class);
+
+        when(user.getPassword()).thenReturn("SomePassword");
+        doNothing().when(user).setEmail(any());
```

Comment:
I'm not sure why doNothing() is needed. Could you maybe explain it?

Reply:
If it's just to something diffrent/nothing, why isn't it possible to just set it to other data or not to call set?

Reply:
It is just to mock a void function

Reply:
Yes, but I don't see why to call a function to do nothing. I mean you act then like the user is completetly empty, right? But this would be a diffrent case in my opinion than mismatching email or password.